vtuber/source.h

```diff
@@ -1,31 +1,39 @@
 #pragma once
 
+#include "verilog/dtype.h"
 #include <fstream>
+#include <optional>
 #include <string>
 #include <vector>
 
 template <typename T> class Source {
 public:
   Source() {}
   virtual ~Source() = default;
-  virtual ::std::vector<T> get() = 0;
+  virtual ::std::optional<::std::vector<T>> get() = 0;
```

Comment:
I think a vector of size 0 is enough? What is the difference of returning vector of size 0 and an invalid optional?

Reply:
I let user set the filename, so the file may be not exists.
Since I will switch to istream, this won't be a problem after the change